File name too long for AWS S3 private content urls

I'm seeing a lot of runtime errors while loading images from S3 which are private. I'm using 0.9.1. The errors look like this:

[HANEKE][ERROR] Failed to write key https://dxzezjm7q2pi.cloudfront.net/video-thumbs/52458b21e27b764f002aad2f/556cab4226070024009a8cd2.jpg?Expires=2064337091&Signature=TLn2VmYEgss5PGIsQZawjnNdT0YWW3vM3leGPdw1OHX3ZENsCAD1KGLk1C1J2x~0IdjiPRrjwv3RKPECiCyzKuIMI4sOBmSivTKO~n9eeSIjs5XRNvlQKI~FlmRRI-lHtE~PxqFXCE2HqIYch1eUM3~OzUz9NDfdwz4fo7hsrgefeHAVC47wQJ-ihratFeTj0IPTlyWbHmPUekDwC-fHkKZTllspkNapGTM6a6AlDataDeu-jcIaZ0cfneaOKfsCCGiHfW0pOam2O1tMoP1d1s0yQZNvI~k4qvwyLO0t2fogt9Y0GvG2nSIWNiq9IUXb1cVs1YasqGykyFuMD9Moiw__&Key-Pair-Id=APKAJGTKTSRVKBYIUQTQ with error Error Domain=NSCocoaErrorDomain Code=514 "The operation couldn’t be completed. (Cocoa error 514.)" UserInfo=0x7fad32613b70 {NSFilePath=/Users/durre/Library/Developer/CoreSimulator/Devices/F76F415E-00C7-4C5F-B165-44F16AD346A6/data/Containers/Data/Application/5BBED228-D215-4BE8-8302-BCDAFF6C0F81/Library/Caches/io.haneke/shared-images/original/cd0c7fca45dbf858b69bbd29be8e82f5.jpg?Expires=2064337091&Signature=TLn2VmYEgss5PGIsQZawjnNdT0YWW3vM3leGPdw1OHX3ZENsCAD1KGLk1C1J2x~0IdjiPRrjwv3RKPECiCyzKuIMI4sOBmSivTKO~n9eeSIjs5XRNvlQKI~FlmRRI-lHtE~PxqFXCE2HqIYch1eUM3~OzUz9NDfdwz4fo7hsrgefeHAVC47wQJ-ihratFeTj0IPTlyWbHmPUekDwC-fHkKZTllspkNapGTM6a6AlDataDeu-jcIaZ0cfneaOKfsCCGiHfW0pOam2O1tMoP1d1s0yQZNvI~k4qvwyLO0t2fogt9Y0GvG2nSIWNiq9IUXb1cVs1YasqGykyFuMD9Moiw__&Key-Pair-Id=APKAJGTKTSRVKBYIUQTQ, NSUnderlyingError=0x7fad32611270 "The operation couldn’t be completed. File name too long"}

--- a/Haneke/String+Haneke.swift
+++ b/Haneke/String+Haneke.swift
@@ -35,7 +35,7 @@ extension String {
     
     func MD5Filename() -> String {
         let MD5String = self.MD5String()
-        let pathExtension = (self as NSString).pathExtension
+        let pathExtension = NSURL(string: self)?.pathExtension ?? (self as NSString).pathExtension
         if pathExtension.characters.count > 0 {
             return (MD5String as NSString).stringByAppendingPathExtension(pathExtension) ?? MD5String
         } else {

